### PR TITLE
Part3.12

### PR DESCRIPTION
**Pull request by moohere**

### CODE REVIEW

File: NewHaven/CMakeLists.txt [hunk 1]
```diff
@@ -13,13 +13,6 @@ if(UNIX)
...
-add_executable(PlayerTest test/PlayerTest.cpp src/Player.cpp src/GBMap.cpp src/Square.cpp src/Resources.cpp src/ResourceScore.cpp src/VGMap.cpp src/ScoreCalculator.cpp)
-target_link_libraries (PlayerTest ${Boost_LIBRARIES})
-target_link_libraries(MapLoaderTest   ${Boost_LIBRARIES})
-target_link_libraries (VGMap ${Boost_LIBRARIES})
-target_link_libraries (GameBoardMapTest ${Boost_LIBRARIES})
-target_link_libraries (PlayerTest ${Boost_LIBRARIES})
\ No newline at end of file
+
+add_executable(TurnSequenceTest test/TurnSequence.cpp src/Player.cpp src/GBMap.cpp src/VGMap.cpp src/Square.cpp src/Resources.cpp src/ResourceScore.cpp src/ScoreCalculator.cpp)
+target_link_libraries (TurnSequenceTest ${Boost_LIBRARIES})
```
Comment (MahChaa) [inline NewHaven/CMakeLists.txt]: Keep the other executables, just add yours.

Comment (MahChaa) [inline NewHaven/CMakeLists.txt]: Add your executable, but don't delete the others yet. We might need them for testing if we make changes to the classes.

File: NewHaven/test/TurnSequence.cpp [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+    // Initiate decks
+    auto *bDeck = new BuildingDeck();
+    auto *hDeck = new HarvestDeck();
+
+    auto *player = new Player(); // Initiate player
+
+    // Start of the game, each player should have 6 buildings and 2 Harvest Tiles
+    for (int i = 0; i < 5; i ++)
+        (*player).drawBuilding(*bDeck->draw()); // Player draws a building from the deck
```
Comment (MahChaa) [inline NewHaven/test/TurnSequence.cpp]: You say each player should have 6 buildings, but proceed to only draw 5 buildings.

File: NewHaven/test/TurnSequence.cpp [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    for (int i = 0; i < 2; i++)
+        (*player).drawHarvestTile(*hDeck->draw()); //Player draws a harvest tile from the deck
+
+    // This tile is kept FACE DOWN until played.
+    HarvestTile *shipmentTilePtr;
+    HarvestTile shipmentTile = *hDeck->draw();
+
+    shipmentTilePtr = &shipmentTile;
```
Comment (MahChaa) [inline NewHaven/test/TurnSequence.cpp]: Hand has a shipment member variable, which is a HarvestTile.

File: NewHaven/test/TurnSequence.cpp [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+        // *IMPORTANT* Requires the map driver for the resource trail
+        //(*player).resourceTracker = (*player).calculateResources();
+
+        // *IMPORTANT*, this part needs to be done at the start of the round
+        // the NEXT time the player plays. This way we can count the resources for
+        // the intiial round, then convert it to normal for the start of the next round.
+        // I'll leave this here for this driver for now.
+        // Point the shipment tile back to the original tile
+        *shipmentTilePtr = shipmentTile;
```
Comment (MahChaa) [inline NewHaven/test/TurnSequence.cpp]: I think this can be much less "hacky" if you use the `shipment` member variable of hand.

**Review by MahChaa — CHANGES_REQUESTED** [review]
Don't delete the other test drivers, they'll still be useful if we make changes to the classes. The way you're implementing the shipment doesn't make use of the shipment member variable of `Hand`, which I think would help you doing what you want. You didn't test the whole thing properly in your driver, We still don't know if it'll work properly with `ResourceTracker`. Let the player places a few harvest tiles including the shipment and the calculate the score.

File: NewHaven/test/TurnSequence.cpp (hunk 3 above)
Comment (moohere) [reply NewHaven/test/TurnSequence.cpp]: Ah I didn't notice that, let me try to change things up according to that then! thanks

**Review by mastrHyperion98 — CHANGES_REQUESTED** [review]
PR hasnt been updated yet, consider merging A2 into the branch to resolve merge conflicts

**Pull request closed without merging**